File: content/atlas/source/includes/fts/lookup-with-search/create-index-example.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_analytics"]
collection = database["accounts"]

# Create the MongoDB Search index definition for the document field
search_index_model = SearchIndexModel(
    definition={
        "mappings": {
            "dynamic": true
        },
    },
    name="lookup-with-search-tutorial",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")

File: content/search/source/includes/index/field-types/string/code-snippets/java/CreateIndexExample.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class IndexExample {
    public static void main(String[] args) {
        // connect to your Atlas cluster
        String uri = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");

            Document index = new Document("mappings",
                    new Document("dynamic", false),
                            .append("similarity": { "type": "bm25" }),
                            .append("fields", new Document("title", new Document("type", "string"))));
            String result = collection.createSearchIndex(index);
            System.out.println("New index name: " + result);
        }
    }
}